sentry-logback/src/main/java/io/sentry/logback/SentryAppender.java

```diff
@@ -0,0 +1,164 @@
+public final class SentryAppender extends UnsynchronizedAppenderBase<ILoggingEvent> {
+  private @Nullable String dsn;
+  private @Nullable String environment;
+  private @Nullable Integer maxBreadcrumbs;
+  private @Nullable Integer shutdownTimeoutMillis;
+  private @Nullable Integer flushTimeoutMillis;
+  private @Nullable Integer readTimeoutMillis;
+  private @Nullable Double sampleRate;
+  private @Nullable Boolean bypassSecurity;
+  private @Nullable Boolean debug;
+  private @Nullable Boolean attachThreads;
+  private @Nullable Boolean attachStacktrace;
+
+  @Override
+  public void start() {
+    if (dsn != null) {
+      Sentry.init(
+          options -> {
+            options.setDsn(dsn);
+            Optional.ofNullable(maxBreadcrumbs).ifPresent(options::setMaxBreadcrumbs);
+            Optional.ofNullable(environment).ifPresent(options::setEnvironment);
+            Optional.ofNullable(shutdownTimeoutMillis).ifPresent(options::setShutdownTimeout);
+            Optional.ofNullable(flushTimeoutMillis).ifPresent(options::setFlushTimeoutMillis);
+            Optional.ofNullable(readTimeoutMillis).ifPresent(options::setReadTimeoutMillis);
+            Optional.ofNullable(sampleRate).ifPresent(options::setSampleRate);
+            Optional.ofNullable(bypassSecurity).ifPresent(options::setBypassSecurity);
+            Optional.ofNullable(debug).ifPresent(options::setDebug);
+            Optional.ofNullable(attachThreads).ifPresent(options::setAttachThreads);
+            Optional.ofNullable(attachStacktrace).ifPresent(options::setAttachStacktrace);
+          });
+    }
+    super.start();
+  }
+
+  @Override
+  protected void append(@NotNull ILoggingEvent eventObject) {
+    if (Sentry.isEnabled()) {
+      Sentry.captureEvent(createEvent(eventObject));
+    }
+  }
+
+  /**
+   * Creates {@link SentryEvent} from Logback's {@link ILoggingEvent}.
+   *
+   * @param loggingEvent the logback event
+   * @return the sentry event
+   */
+  @SuppressWarnings("JdkObsolete")
+  final @NotNull SentryEvent createEvent(@NotNull ILoggingEvent loggingEvent) {
+    final SentryEvent event = new SentryEvent(new Date(loggingEvent.getTimeStamp()));
+    final Message message = new Message();
+    message.setFormatted(loggingEvent.getFormattedMessage());
+    message.setParams(toParams(loggingEvent.getArgumentArray()));
+    event.setMessage(message);
+    event.setLogger(loggingEvent.getLoggerName());
+    event.setLevel(formatLevel(loggingEvent.getLevel()));
+
+    final ThrowableProxy throwableInformation = (ThrowableProxy) loggingEvent.getThrowableProxy();
+    if (throwableInformation != null) {
+      event.setThrowable(throwableInformation.getThrowable());
+    }
+
+    if (loggingEvent.getThreadName() != null) {
+      event.setExtra("thread_name", loggingEvent.getThreadName());
+    }
+
+    for (Map.Entry<String, String> entry : loggingEvent.getMDCPropertyMap().entrySet()) {
+      event.setTag(entry.getKey(), entry.getValue());
```

Comment:
just wondering if we should do `setTag(x, y)` or `setExtras(x, y)` or `event.setContexts.out(x, y)`, because `getMDCPropertyMap` means `map diagnostic context`, the old version add as a tag so maybe we should keep as it is to keep compatibility

Reply:
I think you mean that old version adds as `extras`? I see that in old SDK there was an option to tell Sentry which MDC tags qualify as `tag` and which as `extra`. In 2.0 it's not there anymore so we can either add it or pick `tags` or `extras` I think

Reply:
yeah its trick: https://github.com/getsentry/sentry-java/blob/master/sentry-logback/src/main/java/io/sentry/logback/SentryAppender.java#L172-L178
I'd suggest then to use `setExtras` instead of `setTag`, wdyt?
I don't think we need this configuration to choose when its a tag or not, at least not in the 1st version.

Reply:
What's exactly the difference in practice between tags and extras? I see that they are presented differently and users can filter issues by tags. Extras are just metadata attached to an event but users can't do anything with it. Am I correct?

Reply:
yes, exactly, and looks like getMDCPropertyMap returns only metadata as well

Reply:
But without the ability to search by this metadata its usage is very limited. I imagine users may want to put a `userId` to MDC and expect to search events for specific user.

Perhaps it makes sense to add a property to `SentryAppender` `mdcTags` - similar to get the same functionality as in Sentry SDK 1.x without cluttering `sentry-core`?

Reply:
I get your point, for `user` for example, our protocol has a dedicated field, but with this integration, it'd be hard to know which key is user-related.
I'd say we can start with everything being an extra and improve from there when we get feedback.
@bruno-garcia opinions?

Reply:
Ok I set it as extras 👍